addons/godot-next/resources/behavior.gd

```diff
@@ -0,0 +1,67 @@
+# author: xdgamestudios
+# license: MIT
+# description: This is an abstract base class for all Behaviors. 
+tool
+extends Resource
+class_name Behavior
+
+##### CLASSES #####
+
+##### SIGNALS #####
+
+##### CONSTANTS #####
+
+##### EXPORTS #####
+
+##### PROPERTIES #####
+
+var owner = null
+#warning-ignore:unused_class_variable
+var enabled: bool = false
+
+##### NOTIFICATIONS #####
+
+##### OVERRIDES #####
+
+func _ready() -> void:
+	pass
+
+#warning-ignore:unused_argument
```

Comment:
Now that I think about it, if our Behaviors class is calling the methods *if they are implemented*, this will force every Behavior to have the method implemented. We might actually want to take all of these out and just document at the top in the comments that the option is available. Specify which notifications are supported.

Reply:
Okay will add that to the comments